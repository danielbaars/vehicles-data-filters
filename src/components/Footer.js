import React from 'react';
import PropTypes from 'prop-types';

import Row from './Row';
import FramedBox from './FramedBox';

const Credits = ({ className, text, url, logo }) => {
  return (
    <div className='small-12 medium-6 cell credits'>
      <div className={`credits__${className}-container`}>
        <div className={`credits__${className}`}>
          <div className='credits__text'>{text}</div>
          <a className={`credits__logo credits__logo--${className} imgc`} href={url} target='_blank'>
            <img src={logo} alt={`logo--${className}`} />
          </a>
        </div>
      </div>
    </div>
  );
};

Credits.propTypes = {
  className: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  logo: PropTypes.string.isRequired,
};

const Footer = () => {
  const sytacLogo = 'http://www.sytac.nl/wp-content/themes/sytac/images/sytac_new.png';
  const danielLogo = 'https://rawgit.com/danielbaars/5171605b94645c9e87e154c3486e4ef3/raw/c49b826993d925625ee819d51797ac29d05eba89/daniel-name-logo.svg';
  const sytacUrl = 'http://www.sytac.nl/';
  const danielUrl = 'https://github.com/danielbaars';
  return (
    <div className='footer__outer'>
      <Row>
        <div className='small-12 cell'>
          <FramedBox className='credits'>
            <div className='grid-x grid-margin-x'>
              <Credits className='inspired' text='Inspired by' url={sytacUrl} logo={sytacLogo} />
              <Credits className='built' text='Built by' url={danielUrl} logo={danielLogo} />
            </div>
          </FramedBox>
        </div>
      </Row>
    </div>
  );
};

export default Footer;
