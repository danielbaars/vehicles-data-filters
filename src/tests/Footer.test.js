import React from 'react';
import renderer from 'react-test-renderer';

import Footer from '../components/Footer';

describe('Footer', () => {
  it('renders and matches our snapshot', () => {
    const component = renderer.create(<Footer />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
