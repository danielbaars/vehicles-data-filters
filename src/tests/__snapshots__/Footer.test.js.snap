// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders and matches our snapshot 1`] = `
<div
  className="footer__outer outer"
>
  <div
    className="grid-container"
  >
    <div
      className="grid-x grid-margin-x "
    >
      <div
        className="small-12 cell"
      >
        <div
          className="credits inner"
        >
          <div
            className="credits__box box"
          >
            <div
              className="grid-x grid-margin-x"
            >
              <div
                className="small-12 medium-6 credits cell"
              >
                <div
                  className="credits__inspired-container"
                >
                  <div
                    className="credits__inspired"
                  >
                    <div
                      className="credits__text"
                    >
                      Inspired by
                    </div>
                    <a
                      className="credits__logo credits__logo--inspired imgc"
                      href="http://www.sytac.nl/"
                      target="_blank"
                    >
                      <img
                        alt="logo--inspired"
                        src="http://www.sytac.nl/wp-content/themes/sytac/images/sytac_new.png"
                      />
                    </a>
                  </div>
                </div>
              </div>
              <div
                className="small-12 medium-6 credits cell"
              >
                <div
                  className="credits__built-container"
                >
                  <div
                    className="credits__built"
                  >
                    <div
                      className="credits__text"
                    >
                      Built by
                    </div>
                    <a
                      className="credits__logo credits__logo--built imgc"
                      href="https://github.com/danielbaars"
                      target="_blank"
                    >
                      <img
                        alt="logo--built"
                        src="https://rawgit.com/danielbaars/5171605b94645c9e87e154c3486e4ef3/raw/c49b826993d925625ee819d51797ac29d05eba89/daniel-name-logo.svg"
                      />
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
